### ISSUE TITLE
guildScheduledEventUser Add & Remove not emitting

### ISSUE THREAD

**Issue by Inrixia** [discordjs/discord.js#8146]
### Which package is this bug report for?

discord.js

### Issue description

When attempting to listen for `guildScheduledEventUserAdd` and `guildScheduledEventUserRemove` I am not receiving any events.

I am toggling interested on events that the bot belongs in. Update/Delete/Create works fine.

### Code sample

```typescript
import { Client, Intents } from "discord.js";

// Create a new client instance
const client = new Client({ intents: [Intents.FLAGS.GUILD_SCHEDULED_EVENTS], partials: ["GUILD_SCHEDULED_EVENT"] });

// Login to Discord with your client's token
client.login("OTg2OTc4NjA2MzUxNzg2MDY1.GNz0ua.6EoctC6Q2bOzXzhSUycKgZBEYgxmjxi_h-PUA8");
client.once("ready", () => {
	console.log("o/");
	client
		// .on("guildScheduledEventUpdate", console.log)
		// .on("guildScheduledEventCreate", console.log)
		// .on("guildScheduledEventDelete", console.log)
		.on("guildScheduledEventUserAdd", console.log)
		.on("guildScheduledEventUserRemove", console.log);
});
```

### Package version

^13.8.0

### Node.js version

v17.3.1

### Operating system

Windows

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

GuildScheduledEvent

### Which gateway intents are you subscribing to?

GuildScheduledEvents

**Comment by Inrixia** [comment]
Woops, forgot to remove the token after copy-pasting the code again. I've invalidated it.

**Comment by Jiralite** [comment]
If the events are not firing, then there is only one scenario left: an uncached user.

Line 13 in the extract of code sent above is `const user = this.getUser(data);`. If the user is not cached already by the bot, we exit out on line 15, resulting in the events not being fired.

If you still want these events being fired on uncached users, you will need the `USER` partial.

**Comment by Inrixia** [comment]
Yep! Adding the `USER` partial and `Intents.FLAGS.GUILDS` fixed this. 

Note the `GUILD_SCHEDULED_EVENT` partial does not appear to be needed.

Makes sense, thanks for the help!
